src/send/useFetchRecipientVerificationStatus.ts

```diff
@@ -13,6 +17,8 @@ const useFetchRecipientVerificationStatus = () => {
   )
 
   const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+  const verifiedAddresses = useSelector(verifiedAddressesSelector)
+  const verifiedAddressesLoading = useSelector(verifiedAddressesLoadingSelector)
```

Comment:
;The reason why we need this extra piece of state is kind of subtle, and tripped me up for a bit. Basically, `getRecipientVerificationStatus` (in the `useEffect` hook below) is designed such that it assumes its parameters (`e164NumberToAddress`, `verifiedAddresses`) are not in-flight, i.e., not currently "loading". As such, it returns `RecipientVerificationStatus.VERIFIED` when the address is missing from `verifiedAddresses`, which _will always be the case_ while we're fetching an address we haven't seen before. Because of the `RecipientVerificationStatus.UNKNOWN` check below, we'll never change the status after the fetch finishes, and as soon as the verification status is set to verified, we enable the "Send" button.

What we actually want is to be able to keep the Send button disabled until the fetch is done, then _always_ enable it, but show a banner if the address is unverified, which is what this loading state allows for (only updating the verification status once the fetch is definitely done). @satish-ravi, I think in #4584 you should just be able to look at a combination of `recipientVerificationStatus` and `selectedRecipient.recipientType` and only show the banner if:

```ts
recipientVerificationStatus === RecipientVerificationStatus.UNVERIFIED && selectedRecipient.recipientType === RecipientType.Address
```